src/CDataInterface/format_string_to_types.jl

```diff
@@ -0,0 +1,68 @@
+# https://arrow.apache.org/docs/format/CDataInterface.html#data-type-description-format-strings
+module FormatStrings
+
+function get_type_from_format_string(format_string ::String) ::DataType
+    if format_string == "n"
+        Nothing
+    elseif format_string == "b"
+        Bool
+    elseif format_string == "c"
+        Int8
+    elseif format_string == "C"
+        UInt8
+    elseif format_string == "s"
+        Int16
+    elseif format_string == "S"
+        UInt16
+    elseif format_string == "i"
+        Int32
+    elseif format_string == "I"
+        UInt32
+    elseif format_string == "l"
+        Int64
+    elseif format_string == "L"
+        UInt64
+    elseif format_string == "e"
+        Float16
+    elseif format_string == "f"
+        Float32
+    elseif format_string == "g"
+        Float64
+    elseif format_string == "z" || format_string == "Z"
+        Vector{UInt8}
+    elseif format_string == "u" || format_string == "U"
+        String
+    elseif format_string[1] == 'd'
+        const splits = split(format_string[3:end], ",")
+        precision = Int(splits[1])
+        scale = Int(splits[2])
+        if length(splits) == 3
+            bandwidth = splits[3]
+        end
+        #TODO return something here
+    elseif format_string[1] == 'w'
+        #TODO figure out fixed width binary
```

Comment:
This will just be the same as `Arrow.FixedSizeList`, but with `UInt8` as the element type

Reply:
So a fixed width binary type won't have any children; it's like a hard-coded `UInt8` child type and so doesn't need to be parsed recursively.